src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestChangePolicyAction.kt

```diff
@@ -113,7 +117,20 @@ class RestChangePolicyAction(controller: RestController, val clusterService: Clu
             if (!response.isExists || response.isSourceEmpty) {
                 return channel.sendResponse(BytesRestResponse(RestStatus.NOT_FOUND, "Could not find policy=${changePolicy.policyID}"))
             }
+            this.response = response
+            IndexUtils.updateIndexMapping(INDEX_STATE_MANAGEMENT_INDEX,
```

Comment:
because we do not need to update the mapping if policy does not exist. Only if the user update ISM plugin version and their first call to ISM is to change policy then we want to ensure updating index mapping.